src/test/java/com/galvanize/customer/controllers/CustomerRestControllerTests.java

```diff
@@ -27,47 +33,59 @@
 
     @Autowired
     CustomerRepository repository;
-
+    String jsonPath ;
     @Before
-    public void setUp() throws Exception {
+    public void setUp() {
 
+        //Set up
+        jsonPath = "{\"address\":\"55 Hawthorne Rd\",\"city\":\"Raleigh\",\"dateJoined\":\"12/12/2019\",\"firstName\":\"Eva\",\"lastName" +
+                "\":\"Mendez\"," +
+                "\"phoneNumber\":\"9673019375\",\"state\":\"NC\",\"zip\":\"27603\"}";
     }
 
     @Test
     public void addCustomer() throws Exception {
+        //Execute and Assertion
+        mvc.perform(post("").contentType(MediaType.APPLICATION_JSON)
+        .content(jsonPath))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("customerId").exists());
     }
 
     @Test
     public void getCustomersByState() throws Exception {
+        mvc.perform(get("/State/NC"))
```

Comment:
Should do some spot checking that data exists, AND that only customers from "NC" are returned.

Reply:
I have used .andExpect(jsonPath("$[0].state").value("MN")). Since multiple values are returned. Is there a way to loop through it ?

Reply:
I haven't found a way to loop through it using .andExect().  However, you can extract the json from the stream, and then process it.  Meaning, take the json, and loop through it as a string, or convert it to, in this case, a List<Customer>, and then loop through the list.   Something like this...

MvcResult result = mockMvc.perform(post("/api/users").header("Authorization", base64ForTestUser).contentType(MediaType.APPLICATION_JSON)
            .content("{\"userName\":\"testUserDetails\",\"firstName\":\"xxx\",\"lastName\":\"xxx\",\"password\":\"xxx\"}"))
            .andDo(MockMvcResultHandlers.print())
            .andExpect(status().isBadRequest())
            .andReturn();

String content = result.getResponse().getContentAsString();